docs/content/tutorial/step_04.ngdoc

```diff
@@ -1,199 +1,310 @@
 
-  This is a good time to talk about two-way data-binding. Notice that when the app is loaded in the
-browser, "Newest" is selected in the drop down menu. This is because we set `orderProp` to `'age'`
-in the controller. So the binding works in the direction from our model to the UI. Now if you
-select "Alphabetically" in the drop down menu, the model will be updated as well and the phones
-will be reordered. That is the data-binding doing its job in the opposite direction — from the UI
-to the model.
+Yeah, you guessed it: There is a better way !
 
+Each feature/section, will declare its own module and have all related entities registered there.
+The main module (`phonecatApp`) will declare a dependency on each feature/section module. Now,
+all it takes to reuse the same code on new project is copying the feature directory over and adding
+the feature module as a dependency in the new project's main module.
 
+Here is what our `phoneList` feature will look like after this change:
 
-## Test
+<br />
+**`/`:**
 
-The changes we made should be verified with both a unit test and an end-to-end test. Let's look at
-the unit test first.
+```
+  app/
+    phone-list/
+      phone-list.module.js
+      phone-list.component.js
+      phone-list.component.spec.js
+    app.module.js
+```
 
-__`test/unit/controllersSpec.js`:__
+<br />
+**`app/phone-list/phone-list.module.js`:**
 
 ```js
-describe('PhoneCat controllers', function() {
-
-  describe('PhoneListCtrl', function(){
-    var scope, ctrl;
-
-    beforeEach(module('phonecatApp'));
+// Define the `phoneList` module
+angular.module('phoneList', []);
+```
 
-    beforeEach(inject(function($controller) {
-      scope = {};
-      ctrl = $controller('PhoneListCtrl', {$scope:scope});
-    }));
+<br />
+**`app/phone-list/phone-list.component.js`:**
 
-    it('should create "phones" model with 3 phones', function() {
-      expect(scope.phones.length).toBe(3);
-    });
+```js
+// Register the `phoneList` component on the `phoneList` module,
+angular.
+  module('phoneList').
+  component('phoneList', {...});
+```
 
+<br />
+**`app/app.module.js`:**<br />
+_(since `app/app.js` now only contains the main module declaration, we gave it a `.module` suffix)_
 
-    it('should set the default value of orderProp model', function() {
-      expect(scope.orderProp).toBe('age');
-    });
-  });
-});
+```js
+// Define the `phonecatApp` module
+angular.module('phonecatApp', [
+  // ...which depends on the `phoneList` module
+  'phoneList'
+]);
 ```
 
+By passing `phoneList` inside the dependencies array when defining the `phonecatApp` module, Angular
+will make all entities registered on `phoneList` available on `phonecatApp` as well.
 
-The unit test now verifies that the default ordering property is set.
+<div class="alert alert-info">
+  <p>
+    Don't forget to also update your `index.html` adding a `<script>` tag for each JacaScript file
+    we have created. This might seem tedious, but is totally worth it.
```

Comment:
It might be worth mentioning (since you've done it this way) that the module files must be loaded before component files.